dnb_worldbase/mapping.py

```
@@ -0,0 +1,123 @@
+import logging
+
+from .constants import (BUSINESS_INDICATOR_MAPPING, DNB_COUNTRY_CODE_MAPPING,
+                        LEGAL_STATUS_CODE_MAPPING, NATIONAL_ID_CODE_MAPPING)
+
+
+class DataMappingError(Exception):
+    pass
+
+
+logger = logging.getLogger(__name__)
+
+
+def dnb_country_lookup(dnb_country_code):
+    """
+    Takes a dnb country code and returns the associated iso alpha 2 code.
+
+    It will raise an DataMappingException if: the code does not exist in the country code mapping or if the
+    mapping entry does not have an associated iso alpha 2 code.  Not all DNB country codes map to iso 2 codes.
+    """
+
+    if dnb_country_code == '':
+        return None
+
+    if dnb_country_code not in DNB_COUNTRY_CODE_MAPPING:
+        raise DataMappingError(f'{dnb_country_code} is not in mapping')
+
+    iso_alpha2_code = DNB_COUNTRY_CODE_MAPPING[dnb_country_code]
+
+    if not iso_alpha2_code:
+        raise DataMappingError(f'{dnb_country_code} does not have an associated iso alpha 2 code')
+
+    return iso_alpha2_code
+
+
+def map_legal_status(legal_status_code):
+    """Takes the worldbase legal status code and returns the """
+
+    if legal_status_code == '' or int(legal_status_code) not in LEGAL_STATUS_CODE_MAPPING:
```

A very minor thing, but I would probably just have the check on 
```
if legal_status_code not in LEGAL_STATUS_CODE_MAPPING:
   ...
```
and this would cover all the cases of an unknown legal status code right now, as well as not having any gotchas if a non-integer status code needs to be added later.

---

(Ah, and use strings for the keys in LEGAL_STATUS_CODE_MAPPING)

---

(Actually, probably use strings for _all_ the mapping dicts. Since it's strings that always come out of the CSV reading, and then we map from CSV value -> "our" value, it seems like a bit of an unnecessary step to convert to `int`)

---

Actually, I guess it's typically a bit more Pythonic to follow the ask-for-forgiveness pattern? (Not that being Pythonic is always the way to go, but somehow it does keep things neat when parsing/ingesting things...)
```
try:
  return LEGAL_STATUS_CODE_MAPPING[legal_status_code].name
except KeyError:
  raise DataMappingError(f'no mapping for legal status code: {legal_status_code}')
```

---

Ah! I see this pattern is used in `map_business_indicator`

---

Yep, definitely agreed re's EAFTP.  In fact, I think I may just ditch the 'DataMappingError' exception altogether (if feasible, but it should be) and `except BaseException` upstream.

Also, yes agreed string based options would make things easier.

---

> and except BaseException upstream

Sounds reasonable, but I might only catch the exceptions we expect from bad data, e.g. ValueError, KeyError etc, and let the rest bubble up